File: python/day_02a.py
#!/usr/bin/env python3

import re


def main():
    f = open("../input/day_02_input")
    match_pattern = re.compile(r"(\d+)-(\d+) ([a-z]): (.*)\n")
    count = 0
    for line in f:
        low, high, letter, pwd = re.match(match_pattern, line).groups()
        n = pwd.count(letter)
        if n >= int(low) and n <= int(high):
            count += 1
    print(count)
    return count


if __name__ == "__main__":
    main()

File: 2020/cpp/day_02a.cpp
#include <algorithm>
#include <fstream>
#include <iostream>
#include <unordered_set>

int main() {
  std::ifstream file;
  file.open("../input/day_02_input");
  std::unordered_set<int> entries;
  char dash, letter, colon;
  std::string pwd;
  int low, high;
  int count = 0;
  while (file >> low >> dash >> high >> letter >> colon >> pwd) {
    const int reps =
        std::count_if(std::begin(pwd), std::end(pwd),
                      [=](const auto ele) { return ele == letter; });
    if (low <= reps && reps <= high) {
      ++count;
    }
  }
  std::cout << count << '\n';
  return count;
}
